src/app/public/src/modules/nav/nav.service.ts

```diff
@@ -46,14 +50,59 @@ export class StacheNavService {
     return /^(https?|mailto|ftp):+|^(www)/.test(path);
   }
 
+  public updateRoutesOnScroll(routes: StacheNavLink[]) {
+    if (routes && routes.length) {
+      this.updateView(routes);
+    }
+  }
+
+  // Updates ng-class level trigger to show/hide blue border marking location on page
+  public updateView(routes: StacheNavLink[]) {
+    this.trackPageOffset();
+    this.isCurrent(routes);
+  }
+
+  // Checks to see if stache-wrapper exists and provides offset from top of window
+  private getHeaderHeight() {
+    const stacheWrapper = this.windowRef.nativeWindow.document.querySelector('.stache-wrapper');
+
+    if (stacheWrapper) {
+      this.headerHeight = stacheWrapper.offsetTop;
+    }
+  }
+
+  private trackPageOffset() {
+    this.getHeaderHeight();
+
+    // Represents top of page
+    // headerHeight buffer removes discrepency when hero or other 'header' elements exist
+    // 50px buffer aids in click-to-anchor and user visualization
+    this.pageOffset = ((this.windowRef.nativeWindow.pageYOffset - this.headerHeight) + 50);
```

Comment:
Can we rename this variable to make it more self descriptive and remove the comments?